cli/data.go

```diff
@@ -132,20 +196,28 @@ func createDataFilter(c *cli.Context) (*datapb.Filter, error) {
 	if len(c.StringSlice(dataFlagMimeTypes)) != 0 {
 		filter.MimeType = c.StringSlice(dataFlagMimeTypes)
 	}
+	if len(c.StringSlice(dataFlagFilterTags)) != 0 {
+		filter.TagsFilter = &datapb.TagsFilter{
+			Type: datapb.TagsFilterType_TAGS_FILTER_TYPE_MATCH_BY_OR,
+			Tags: c.StringSlice(dataFlagFilterTags),
+		}
+	}
 	if c.StringSlice(dataFlagTags) != nil {
-		switch {
-		case len(c.StringSlice(dataFlagTags)) == 1 && c.StringSlice(dataFlagTags)[0] == "tagged":
-			filter.TagsFilter = &datapb.TagsFilter{
-				Type: datapb.TagsFilterType_TAGS_FILTER_TYPE_TAGGED,
-			}
-		case len(c.StringSlice(dataFlagTags)) == 1 && c.StringSlice(dataFlagTags)[0] == "untagged":
-			filter.TagsFilter = &datapb.TagsFilter{
-				Type: datapb.TagsFilterType_TAGS_FILTER_TYPE_UNTAGGED,
-			}
-		default:
-			filter.TagsFilter = &datapb.TagsFilter{
-				Type: datapb.TagsFilterType_TAGS_FILTER_TYPE_MATCH_BY_OR,
-				Tags: c.StringSlice(dataFlagTags),
+		if len(c.StringSlice(dataFlagFilterTags)) == 0 {
```

Comment:
Can you add a comment here explaining the conditional logic (what happens when --filter-tag is set versus not set)? I found that it took me some time to work out what was exactly happening here

Reply:
Certainly! Thanks